cmake/Modules/ObsDefaults_Windows.cmake

```diff
@@ -0,0 +1,123 @@
+cmake_minimum_required(VERSION 3.20)
+
+if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+  set(CMAKE_INSTALL_PREFIX
+      "${CMAKE_BINARY_DIR}/rundir"
+      CACHE STRING "Directory to install OBS after building" FORCE)
+endif()
+
+# Enable building Windows modules with file descriptors
+# https://github.com/obsproject/obs-studio/commit/51be039cf82fc347587d16b48f74e65e86bee301
+set(MODULE_DESCRIPTION "OBS Studio")
+
+macro(setup_obs_project)
+  if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+    set(_ARCH_SUFFIX 64)
+  else()
+    set(_ARCH_SUFFIX 32)
+  endif()
+
+  set(OBS_OUTPUT_DIR "${CMAKE_BINARY_DIR}/rundir")
+
+  set(OBS_EXECUTABLE_DESTINATION "bin/${_ARCH_SUFFIX}bit")
+  set(OBS_EXECUTABLE32_DESTINATION "bin/32bit")
+  set(OBS_EXECUTABLE64_DESTINATION "bin/64bit")
+  set(OBS_LIBRARY_DESTINATION "bin/${_ARCH_SUFFIX}bit")
+  set(OBS_LIBRARY32_DESTINATION "bin/32bit")
+  set(OBS_LIBRARY64_DESTINATION "bin/64bit")
+  set(OBS_PLUGIN_DESTINATION "obs-plugins/${_ARCH_SUFFIX}bit")
+  set(OBS_PLUGIN32_DESTINATION "obs-plugins/32bit")
+  set(OBS_PLUGIN64_DESTINATION "obs-plugins/64bit")
+
+  set(OBS_INCLUDE_DESTINATION "include")
+  set(OBS_CMAKE_DESTINATION "cmake")
+  set(OBS_DATA_DESTINATION "data")
+  set(OBS_DATA_PATH "../../${OBS_DATA_DESTINATION}")
+  set(OBS_INSTALL_PREFIX "")
+
+  set(OBS_SCRIPT_PLUGIN_DESTINATION
+      "${OBS_DATA_DESTINATION}/obs-scripting/${_ARCH_SUFFIX}bit")
+  set(OBS_SCRIPT_PLUGIN_PATH "../../${OBS_SCRIPT_PLUGIN_DESTINATION}")
+
+  string(REPLACE "-" ";" UI_VERSION_SPLIT ${OBS_VERSION})
+  list(GET UI_VERSION_SPLIT 0 UI_VERSION)
+  string(REPLACE "." ";" UI_VERSION_SEMANTIC ${UI_VERSION})
+  list(GET UI_VERSION_SEMANTIC 0 UI_VERSION_MAJOR)
+  list(GET UI_VERSION_SEMANTIC 1 UI_VERSION_MINOR)
+  list(GET UI_VERSION_SEMANTIC 2 UI_VERSION_PATCH)
+
+  if(INSTALLER_RUN AND NOT DEFINED ENV{obsInstallerTempDir})
+    message(
+      FATAL_ERROR
+        "Environment variable obsInstallerTempDir is needed for multiarch installer generation"
+    )
+  endif()
+
+  if(DEFINED ENV{DepsPath${_ARCH_SUFFIX}})
+    set(DepsPath${_ARCH_SUFFIX} "$ENV{DepsPath${_ARCH_SUFFIX}}")
+  elseif(DEFINED ENV{DepsPath})
+    set(DepsPath "$ENV{DepsPath}")
+  endif()
+
+  if(DEFINED ENV{QTDIR${_ARCH_SUFFIX}})
+    set(QTDIR${_ARCH_SUFFIX} "$ENV{QTDIR${_ARCH_SUFFIX}}")
+  elseif(DEFINED ENV{QTDIR})
+    set(QTDIR "$ENV{QTDIR}")
+  endif()
```

Comment:
I guess my thinking was to give the more specific variant precedence and if that is not available fall back to the less generic one, also avoiding adding all variants to the prefix path. No strong opinion here as well.

Reply:
I meant more that you do this here:
```
  if(DEFINED ENV{DepsPath${_ARCH_SUFFIX}})
    set(DepsPath${_ARCH_SUFFIX} "$ENV{DepsPath${_ARCH_SUFFIX}}")
  elseif(DEFINED ENV{DepsPath})
    set(DepsPath "$ENV{DepsPath}")
  endif()
```
and then later do
```
  if(DEFINED DepsPath${_ARCH_SUFFIX})
    list(APPEND CMAKE_PREFIX_PATH "${DepsPath${_ARCH_SUFFIX}}"
         "${DepsPath${_ARCH_SUFFIX}}/bin")
  elseif(DEFINED DepsPath)
    list(APPEND CMAKE_PREFIX_PATH "${DepsPath}" "${DepsPath}/bin")
  endif()
```

Why not just:
```
  if(DEFINED ENV{DepsPath${_ARCH_SUFFIX}})
    list(APPEND CMAKE_PREFIX_PATH "$ENV{DepsPath${_ARCH_SUFFIX}}"
         "$ENV{DepsPath${_ARCH_SUFFIX}}/bin")
  elseif(DEFINED ENV{DepsPath})
    list(APPEND CMAKE_PREFIX_PATH "$ENV{DepsPath}" "$ENV{DepsPath}/bin")
  endif()
```

That's the intermediate assignment I'm talking about.

Reply:
If `DepsPath` is set as environment variable but also as CMake cache variable, it would lead to both entries being added to the prefix path. If I set the local variable to the environment variable (if available) it will lead to this quirky situation where the cache variable is not overwritten but overridden in the current scope, thus it will be set only once.

Reply:
I'm not sure I follow about the quirky behavior.  It works fine now though, so I'm not sure I want to fuss with it.